frontend/src/components/profileForm.tsx

```diff
@@ -0,0 +1,146 @@
+import background from "../assets/bgHero.png";
+import {IUser} from "../types.ts";
+import {useState} from "react";
+import axios, {AxiosError} from "axios";
+import toast from "react-hot-toast";
+
+interface ProfileForm{
+    user : IUser | null,
+    dismiss : () => void
+}
+
+export function ProfileForm({user, dismiss} : ProfileForm ){
+    const [email, setEmail] = useState(user?.email)
+    const [username, setName] = useState(user?.username)
+    const [error, setError] = useState({
+        username: "",
+        email: "",
+        message : ""
+    });
+
+    async function updateUser(e : React.FormEvent<HTMLFormElement>){
+        e.preventDefault()
+        try {
+            const updatedUser = {
+                email,
+                username
+            }
+            if(!email){
+                toast.error("Email cannot be empty")
+            }
+            if(!username){
+                toast.error("Username cannot be empty")
+            }
+            const response = await axios.put(`/api/v1/user/update/${user?.id}`,updatedUser);
+            toast.success(response.data.message)
+            dismiss()
+        }catch (e){
+            const axiosError = e as AxiosError<{
```

Comment:
Avoid `as` whenever possible as it does a force conversion to the specified type. 

Potential solution is: 
```suggestion
            const axiosError: AxiosError<{ error: { message: string; }}> = e;
```
You would need to test if it directly works also if you fell the error is commonly used, you can create a type and pass directly. 

For instance: 
```ts
type ErrorType = { error: { message: string; }};
const axiosError: AxiosError<ErrorType> = e;
```